File: backend/internal/domain/vehicle.go
package domain

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Vehicle struct {
	ID           uuid.UUID `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id" form:"id"`
	CompanyID    uuid.UUID `gorm:"type:uuid;not null" json:"company_id" form:"company_id"`
	BranchID     *uuid.UUID `gorm:"type:uuid" json:"branch_id" form:"branch_id"`
	Name         string    `gorm:"type:varchar(255);not null" json:"name" form:"name"`
	LicensePlate string    `gorm:"type:varchar(20);uniqueIndex;not null" json:"license_plate" form:"license_plate"`
	Model        string    `gorm:"type:varchar(100)" json:"model" form:"model"`
	Type         string    `gorm:"type:varchar(50)" json:"type" form:"type"` // Truck, Car, Van, Motorcycle
	Status       string    `gorm:"type:varchar(20);default:'AVAILABLE'" json:"status" form:"status"` // AVAILABLE, IN_USE, MAINTENANCE, BROKEN
	Year         int       `gorm:"type:int" json:"year" form:"year"`
	Mileage      int       `gorm:"type:int;default:0" json:"mileage" form:"mileage"`
	Capacity     float64   `gorm:"type:decimal(10,2);default:0" json:"capacity" form:"capacity"`
	ImageURL     string    `gorm:"type:varchar(255)" json:"image_url" form:"image_url"`
	CreatedAt    time.Time `gorm:"default:now()" json:"created_at"`
	UpdatedAt    time.Time `gorm:"default:now()" json:"updated_at"`

	Company *Company `gorm:"foreignKey:CompanyID" json:"company,omitempty"`
	Branch  *Branch  `gorm:"foreignKey:BranchID" json:"branch,omitempty"`
}

func (v *Vehicle) BeforeCreate(tx *gorm.DB) (err error) {
	if v.ID == uuid.Nil {
		v.ID = uuid.New()
	}
	return
}

File: frontend-web/src/stores/vehicle.ts
import { defineStore } from 'pinia'
import { vehicleApi } from '@/api/vehicle'

export interface Vehicle {
  id: string
  company_id: string
  branch_id?: string
  name: string
  license_plate: string
  model: string
  type: string
  status: 'AVAILABLE' | 'IN_USE' | 'MAINTENANCE' | 'BROKEN'
  year: number
  mileage: number
  capacity: number
  image_url?: string
  branch?: {
    id: string
    name: string
  }
}

export const useVehicleStore = defineStore('vehicle', {
  state: () => ({
    vehicles: [] as Vehicle[],
    logs: [] as any[],
    loading: false,
    error: null as string | null,
  }),

  actions: {
    async fetchVehicles() {
      this.loading = true
      try {
        const response = await vehicleApi.getVehicles()
        this.vehicles = response.data
      } catch (err: any) {
        this.error = err.response?.data?.error || 'Failed to fetch vehicles'
      } finally {
        this.loading = false
      }
    },

    async addVehicle(data: FormData) {
      try {
        await vehicleApi.createVehicle(data)
        await this.fetchVehicles()
      } catch (err: any) {
        throw err.response?.data?.error || 'Failed to add vehicle'
      }
    },

    async updateVehicle(id: string, data: FormData) {
      try {
        await vehicleApi.updateVehicle(id, data)
        await this.fetchVehicles()
      } catch (err: any) {
        throw err.response?.data?.error || 'Failed to update vehicle'
      }
    },

    async removeVehicle(id: string) {
      try {
        await vehicleApi.deleteVehicle(id)
        await this.fetchVehicles()
      } catch (err: any) {
        throw err.response?.data?.error || 'Failed to delete vehicle'
      }
    },

    async fetchVehicleLogs(id: string) {
      try {
        const response = await vehicleApi.getVehicleLogs(id)
        this.logs = response.data
      } catch (err: any) {
        console.error('Failed to fetch vehicle logs:', err)
      }
    }
  }
})
